client/src/Makefile.am

```diff
@@ -36,11 +49,56 @@ CLIENT_COMMON_LDFLAGS = \
   $(MARGO_LDFLAGS)
 
 CLIENT_COMMON_LIBADD = \
-  $(top_builddir)/common/src/libunifyfs_common.la \
+  $(OPT_LIBS) \
   $(MARGO_LIBS) \
-  -lrt -lpthread
+  -lm -lrt -lcrypto -lpthread
+
+LIBCOMMON_SRCS = \
```

Comment:
I think this breaks the `__FILE__` macro in log printing in the client. The pathnames are not correctly cut:

```
2021-01-14T14:02:42 tid=118463 @ invoke_client_mount_rpc() [client.c:270] invoking the mount rpc function in client
2021-01-14T14:02:42 tid=118463 @ invoke_client_mount_rpc() [client.c:287] Got response ret=0
2021-01-14T14:02:42 tid=118463 @ invoke_client_mount_rpc() [client.c:297] My client id is 33
2021-01-14T14:02:42 tid=118463 @ get_page_size() [common/src/unifyfs_logio.c:63] returning page size 65536 B
2021-01-14T14:02:42 tid=118463 @ unifyfs_init() [s.c:1754] FD limit for system = 4096
2021-01-14T14:02:42 tid=118463 @ init_superblock_structures() [s.c:1542] Meta-stacks initialized
```

Reply:
Looks like the `common` dir is expanded as `/autofs/nccs-svm1_techint/home/hs2/projects/UnifyFS/client/src/../../common/src/unifyfs_log.c`. Because of `/client/src../..`, `unifyfs_log_print` cuts extra characters from the pathname.

Reply:
Yeah, I noticed that effect too. I will fix it.